common/cards/alter-egos-iii/hermits/kingjoel-rare.ts

```diff
@@ -45,52 +45,55 @@ class KingJoelRare extends Card {
 	) {
 		const {player} = component
 
-		const firstPickCondition = query.every(
-			query.slot.opponent,
-			query.not(query.slot.active),
-			query.slot.item,
-			query.not(query.slot.empty),
-		)
-
 		const secondPickCondition = query.every(
 			query.slot.currentPlayer,
 			query.not(query.slot.active),
 			query.slot.item,
 			query.slot.empty,
-			query.slot.row(query.row.hasHermit),
+			query.slot.row(query.row.hasHermit, (_game, row) => !!row.health),
 		)
 
-		let fistPickedCard: CardComponent | null = null
+		let firstPickedCard: CardComponent | null = null
 
 		observer.subscribe(player.hooks.onAttack, (attack) => {
 			if (!attack.isAttacker(component.entity) || attack.type !== 'secondary')
 				return
-			if (!game.components.exists(SlotComponent, firstPickCondition)) return
-			if (!game.components.exists(SlotComponent, secondPickCondition)) return
+			const firstPickCondition = query.every(
+				query.slot.opponent,
+				query.not(query.slot.rowIs(player.opponentPlayer.activeRowEntity)), // Specified in case Knockback or Egg is used
+				query.slot.item,
+				query.not(query.slot.empty),
+				query.slot.row((_game, row) => !!row.health),
+			)
```

Comment:
I don't think the comment is necessary here
```suggestion
			const firstPickCondition = query.every(
				query.slot.opponent,
				query.not(query.slot.active),
				query.slot.item,
				query.not(query.slot.empty),
				query.slot.row((_game, row) => !!row.health),
			)
```